libs/platform/src/lib/components/approval-flow/interfaces/approval-data-source.ts

```diff
@@ -0,0 +1,43 @@
+import { Observable } from 'rxjs';
+import { ApprovalProcess } from './approval-process';
+import { ApprovalNode } from './approval-node';
+import { ApprovalUser } from './approval-user';
+
+export interface ApprovalDataSource {
+
+    /**
+     * Fetch of approval process data.
+     */
+    fetch(): Observable<ApprovalProcess>;
+
+    /**
+     * Fetch of user data.
+     */
+    fetchUser(id: string): Observable<any>;
+
+    /**
+     * Update watcher list. Called whenever there is a change
+     * to the watcher list.
+     */
+    updateWatchers(watchers: ApprovalUser[]): void;
```

Comment:
Same query as above, how a datasource with abstract functions helping in workflow

Reply:
answered above